### ISSUE TITLE
Using the jira/jira_issue role/directive as a substitution

### ISSUE THREAD

**Issue by Renopph** [sphinx-contrib/confluencebuilder#601]
Hi,

I attempted to use both the jira role and jira_issue directive in a substitution, but none of the following work:

``.. |<PROJECT-KEY>-<ID>| :jira:`<PROJECT-KEY>-<ID>` ``
``.. |<PROJECT-KEY>-<ID>| replace:: :jira:`<PROJECT-KEY>-<ID>` ``
`.. |<PROJECT-KEY>-<ID>| replace:: .. jira_issue:: <PROJECT-KEY>-<ID>`
`.. |<PROJECT-KEY>-<ID>| jira_issue:: <PROJECT-KEY>-<ID>`

All of these give me the "WARNING: Substitution definition `<PROJECT-KEY>-<ID>` empty or invalid."

I tried both the :jira: and jira_issue:: directly in my documentation, and that works as expected.
When substituting with other directives, the substitution works, e.g. for:

.. | test | image:: test.png

Using | test | in my documentation properly replaces it with the image

Is that the intended behaviour?

I am using confluencebuilder version 1.7.1
sphinx version 4.3.1
Python 3.10.1

This is the report as generated by sphinx-build-confluence report:
```
(system)
 platform: Windows-10-10.0.19044-SP0
   python: 3.10.1 (tags/v3.10.1:2cd268a, Dec 6 2021, 19:10:37) [MSC v.1929 64 bit (AMD64)]
   sphinx: 4.3.1
 requests: 2.26.0
  builder: 1.7.1

(configuration)
confluence_ask_password: True
confluence_page_hierarchy: True
confluence_parent_page: (set)
confluence_publish: True
confluence_publish_dryrun: False
confluence_server_pass: (set)
confluence_server_url: https://<removed>
confluence_server_user: (set)
confluence_space_key: (set; user)

(confluence instance)
 connected: yes
   fetched: yes
   decoded: yes
    parsed: yes
      type: confluence
   version: 7.1.0
     build: 11042
```

**Comment by jdknight** [comment]
This is a limitation of the existing node implementation in this extension. A series of nodes are not properly marked with correct element types nor properly handle copying parameters when copied through the substitution implementation.

I have a prospect fix for this, but just want to perform additional testing to verify some changes.

**Comment by jdknight** [comment]
Thanks for reporting this issue. Changes have been added in #604 to support JIRA directives/role using substitutions. This change has been added into the main branch and should be made available next stable release (v1.8). Users wanting to use substitutions on JIRA directives/roles before a release is made, the [development version](https://sphinxcontrib-confluencebuilder.readthedocs.io/en/stable/install/#development-installation) of this extension can be installed.

**Comment by jdknight** [comment]
v1.8 is now available on PyPI -- marking as closed.
